src/Controls/StreetControls.js

```diff
@@ -1,17 +1,58 @@
+import TWEEN from '@tweenjs/tween.js';
 import * as THREE from 'three';
+import { MAIN_LOOP_EVENTS } from 'Core/MainLoop';
 import FirstPersonControls from 'Controls/FirstPersonControls';
-import AnimationPlayer, { AnimatedExpression } from 'Core/AnimationPlayer';
 
-// Expression used to damp camera's moves
-function moveCameraExp(root, progress) {
-    const dampingProgress = 1 - Math.pow((1 - (Math.sin((progress - 0.5) * Math.PI) * 0.5 + 0.5)), 2);
-    root.camera.position.lerpVectors(root.positionFrom, root.positionTo, dampingProgress);
+const material = new THREE.MeshBasicMaterial({
+    side: THREE.DoubleSide,
+    color: 0xffffff,
+    depthTest: false,
+    transparent: true,
+    opacity: 0.5 });
+
+function createCircle() {
+    const geomCircle = new THREE.CircleGeometry(1, 32);
+    return new THREE.Mesh(geomCircle, material);
+}
+
+function createRectangle() {
+    const geomPlane = new THREE.PlaneGeometry(4, 2, 1);
+    const rectangle = new THREE.Mesh(geomPlane, material);
+    rectangle.rotateX(Math.PI / 2);
+    return rectangle;
+}
+
+// create surfaces, a circle and a rectangle used to help end user
+function initSurfaces(circle = createCircle(), rectangle = createRectangle()) {
+    rectangle.visible = false;
+    circle.visible = false;
+
+    // surface object is used to store the surfaces
+    const surfaces = new THREE.Object3D();
+    surfaces.add(rectangle);
+    surfaces.add(circle);
+
+    // to acces specific surface, we also add it to the surfaces object
```

Comment:
The problem is I don't know the Id given by ThreeJs, and since we can add a custom Mesh, I can't sure of the name of the Mesh.
I change the code to add these meshes as properties of this StreetControls.

Reply:
Good point, nvm then